data/pagination.go

```diff
@@ -192,7 +200,20 @@ func getPageURL(query string, page int, controllerQuery url.Values) (pageURL str
 	}
 
 	// The pageURL is structured so that search query comes first and current page is mentioned last to make it more readable and logical
-	pageURL = "/search?" + queryParam + filterSortLimitParams + pageParam
+	pageQueryString = req.URL.Path + "?" + queryString + filterSortLimitParams + pageParam
 
-	return pageURL
+	return pageQueryString
+}
+
+func buildQueryString(query, populationTypes, dimensions string) string {
+	var u url.URL
+	q := u.Query()
+	q.Set("q", query)
+	if len(populationTypes) > 0 {
+		q.Set("population_types", populationTypes)
+	}
+	if len(dimensions) > 0 {
+		q.Set("dimensions", dimensions)
+	}
+	return q.Encode()
```

Comment:
Is this extra separation necessary for populationTypes and dimensions - given that it already happens for contentTypes and topics? Is the purpose of it to ensure the q comes first and the page comes last?

Reply:
Sort of.  

This is a hack because we're passing strings of text around rather than the query.

The main aim is to ensure the URL would be encoded with the correct '?' & '&' characters, in the case that anything was omitted.  I'm not sure if the 'q' can / will ever be omit-able.  Probably the "correct" fix is to refactor the entire thing, but it's a bunch of work that's not really part of the ticket.